data/actions/scripts/tools/pick.lua

```diff
@@ -1,6 +1,23 @@
 local groundIds = {354, 355}
 
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	if target.itemid == 11227 then -- shiny stone refining
+		local chance = math.random(1,100)
+
+		if chance == 1 then
+			player:addItem(2160, 1) -- 1% chance of getting crystal coin
```

Comment:
I think it would be better if you created a *local* variable to hold the appropriate item id and then add it to the player (writing ``player:addItem`` only once).

You could use some of the available named item ids constants here, e.g. ``ITEM_CRYSTAL_COIN``, ``ITEM_GOLD_COIN``, etc.